django/core/cache/backends/db.py

```diff
@@ -49,52 +49,78 @@ class DatabaseCache(BaseDatabaseCache):
     pickle_protocol = pickle.HIGHEST_PROTOCOL
 
     def get(self, key, default=None, version=None):
-        key = self.make_key(key, version=version)
-        self.validate_key(key)
+        return self.get_many([key], version).get(key, default)
+
+    def get_many(self, keys, version=None):
+        key_map = {self.make_key(key, version=version): key for key in keys}
         db = router.db_for_read(self.cache_model_class)
         connection = connections[db]
         quote_name = connection.ops.quote_name
         table = quote_name(self._table)
 
+        if not keys:
+            return dict()
+
         with connection.cursor() as cursor:
+            if len(key_map) == 1:
+                format_string = '= %s'
+            else:
+                format_string = 'IN(%s' + ', %s' * (len(key_map) - 1) + ')'
```

Comment:
This can be simplified and put in place of `format_string` below:
```python
'= %s' if len(key_map) == 1 else 'IN(%s)' % ', '.join(['%s'] * len(key_map))
```